cmd/dogstatsd/main_windows.go

```diff
@@ -6,19 +6,370 @@
+				// of the entries require strings.
+				if enabled {
+					switch cfg {
+					case "logs_enabled":
+						overrides[cfg] = true
+					case "apm_config.enabled":
+						overrides[cfg] = true
+					case "process_config.enabled":
+						overrides[cfg] = "true"
+					}
+					log.Debugf("Setting %s to true", cfg)
+				} else {
+					switch cfg {
+					case "logs_enabled":
+						overrides[cfg] = false
+					case "apm_config.enabled":
+						overrides[cfg] = false
+					case "process_config.enabled":
+						overrides[cfg] = "disabled"
+					}
+					log.Debugf("Setting %s to false", cfg)
+				}
+			} else {
+				log.Warnf("Unknown setting %s = %s", key, val)
+			}
+		}
+	}
+	if val, _, err = k.GetStringValue("proxy_host"); err == nil && val != "" {
+		var u *url.URL
+		if u, err = url.Parse(val); err != nil {
+			log.Warnf("unable to import value of settings 'proxy_host': %v", err)
+		} else {
+			// set scheme if missing
+			if u.Scheme == "" {
+				u, _ = url.Parse("http://" + val)
+			}
+			if val, _, err = k.GetStringValue("proxy_port"); err == nil && val != "" {
+				u.Host = u.Host + ":" + val
+			}
+			if user, _, _ := k.GetStringValue("proxy_user"); err == nil && user != "" {
+				if pass, _, _ := k.GetStringValue("proxy_password"); err == nil && pass != "" {
+					u.User = url.UserPassword(user, pass)
+				} else {
+					u.User = url.User(user)
+				}
+			}
+		}
+		proxyMap := make(map[string]string)
+		proxyMap["http"] = u.String()
+		proxyMap["https"] = u.String()
+		overrides["proxy"] = proxyMap
+	} else {
+		log.Debug("proxy key not found, not setting proxy config")
+	}
+	if val, _, err = k.GetStringValue("site"); err == nil && val != "" {
+		overrides["site"] = val
+		log.Debugf("Setting site to %s", val)
+	}
+	if val, _, err = k.GetStringValue("dd_url"); err == nil && val != "" {
+		overrides["dd_url"] = val
+		log.Debugf("Setting dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("logs_dd_url"); err == nil && val != "" {
+		overrides["logs_config.logs_dd_url"] = val
+		log.Debugf("Setting logs_config.dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("process_dd_url"); err == nil && val != "" {
+		overrides["process_config.process_dd_url"] = val
+		log.Debugf("Setting process_config.process_dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("trace_dd_url"); err == nil && val != "" {
+		overrides["apm_config.apm_dd_url"] = val
+		log.Debugf("Setting apm_config.apm_dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("py_version"); err == nil && val != "" {
+		overrides["python_version"] = val
+		log.Debugf("Setting python version to %s", val)
+	}
+
+	// apply overrides to the config
+	config.AddOverrides(overrides)
+
+	// build the global agent configuration
+	err = setupConfigWithoutSecrets("")
+	if err != nil {
+		return fmt.Errorf("unable to set up global agent configuration: %v", err)
+	}
+
+	// dump the current configuration to datadog.yaml
+	b, err := yaml.Marshal(config.Datadog.AllSettings())
```

Comment:
🤔 not sure I love the idea of overwriting the user-provided config - it might stump the user; this would also result in removing potential comments users may have added for self-reference in the config, etc. I guess the idea is to get rid of the registry keys, and we thus need to add those overrides elsewhere. What if we just keep the keys around and just override when the keys are found, making it very obvious in the logs that there's a config override taking place?

I'm sure there's a reason we're doing it this way, just want to make sure we absolutely need to overwrite the config. Alternatively, we could perhaps dump the overrides to another yaml, and do it that way? I think we should discuss this.

Reply:
Line 187 above.... same logic as core agent.  We accept the command line options IFF there's not an API key already; if there's an API key, we assume that the config file in place is the "master" version and leave it alone.